### PR TITLE
Add NER plugin to neural chat

### PR DESCRIPTION
**Pull request by letonghan**
## Type of Change

feature

## Description

Add NER plugin to neural chat, including plugin codes, configurations and Readme file.

## How has this PR been tested?

Local test.

## Dependency Change?

No.

### CODE REVIEW

File: intel_extension_for_transformers/neural_chat/pipeline/plugins/ner/ner.py [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+                inplace=True,
+                level="O1",
+                auto_kernel_selection=True,
+            )
+            for i in range(3):
+                self.model.generation_config.pad_token_id = self.tokenizer.pad_token_id = i
+        print("[NER info] Spacy and LLM model initialized.")
+
+
+    def inference(self, query: str, prompt: str=None, max_new_tokens: int=32, temperature: float=0.01, top_k: int=3, repetition_penalty: float=1.1):
```
Comment (lvliang-intel) [inline intel_extension_for_transformers/neural_chat/pipeline/plugins/ner/ner.py:62]: Should add pre_llm_inference_actions interface for plugin.

File: intel_extension_for_transformers/neural_chat/pipeline/plugins/ner/ner.py [hunk 2]
```diff
@@ -0,0 +1,126 @@
+from threading import Thread
```
Comment (lvliang-intel) [inline intel_extension_for_transformers/neural_chat/pipeline/plugins/ner/ner.py:18]: All python code needs to add license declaration.

File: intel_extension_for_transformers/neural_chat/pipeline/plugins/ner/utils/utils.py [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import re
+import os
+import datetime
+from datetime import timezone, timedelta
+
+
+def construct_default_prompt(query: str, cur_time: str) -> str:
```
Comment (XuhuiRen) [inline intel_extension_for_transformers/neural_chat/pipeline/plugins/ner/utils/utils.py:24]: The prompt should follow the same usage as in https://github.com/intel/intel-extension-for-transformers/blob/main/intel_extension_for_transformers/neural_chat/prompts/prompt.py

**Review by lvliang-intel — APPROVED** [review]

**Comment by VincyZhang** [comment]
FAIL: test_get_default_conv_template_v1 (__main__.TestNeuralChatModel)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/intel-extension-for-transformers/intel_extension_for_transformers/neural_chat/tests/api/test_model.py", line 91, in test_get_default_conv_template_v1
    self.assertEqual(text, str(result))
AssertionError: "Conv[27 chars]7b-v1.1', system_template='<|im_start|>system\[543 chars] 0])" != "Conv[27 chars]7b-v1-1', system_template='<|im_start|>system\[543 chars] 0])"
Diff is 1327 characters long. Set self.maxDiff to None to see it.

**Comment by letonghan** [comment]
> ## FAIL: test_get_default_conv_template_v1 (**main**.TestNeuralChatModel)
> Traceback (most recent call last): File "/intel-extension-for-transformers/intel_extension_for_transformers/neural_chat/tests/api/test_model.py", line 91, in test_get_default_conv_template_v1 self.assertEqual(text, str(result)) AssertionError: "Conv[27 chars]7b-v1.1', system_template='<|im_start|>system[543 chars] 0])" != "Conv[27 chars]7b-v1-1', system_template='<|im_start|>system[543 chars] 0])" Diff is 1327 characters long. Set self.maxDiff to None to see it.

Yuxiang is fixing this UT bug.

**Review by hshen14 — APPROVED** [review]

**Pull request merged**